File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        forward = [0 for i in range(n + 1)]
        backward = [0 for i in range(n + 1)]

        for i in range(1, n + 1):
            forward[i] = forward[i - 1] + i

        backward[n] = n

        for i in range(n - 1, 0, -1):
            backward[i] = backward[i + 1] + i

        for i in range(1, n + 1):
            if forward[i] == backward[i]:
                return i

        return -1

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        int[] forward = new int[n + 1];
        int[] backward = new int[n + 1];

        for (int i = 1; i < n + 1; i++) {
            forward[i] = forward[i - 1] + i;
        }

        backward[n] = n;

        for (int i = n - 1; i > 0; i--) {
            backward[i] = backward[i + 1] + i;
        }

        for (int i = 1; i < n + 1; i++) {
            if (forward[i] == backward[i]) {
                return i;
            }
        }

        return -1;
    }
}